modules/runner/src/main/ts/runner/TestRun.ts

```diff
@@ -101,14 +132,33 @@ export const runSuite = (suite: Suite, state: RunState, actions: RunActions, rep
     const runAfterHooks = <T>(error: boolean) => (result: T): Promise<T> =>
       Hooks.runAfter(suite).then(() => error ? Promise.reject(result) : Promise.resolve(result));
 
+    const checkSiblings = !suite.root ? undefined :
+      () => {
+        const siblings = state.checkSiblings();
+        if (siblings.length > 0) {
+          return fakeFailure(reporter, suite, state, siblings, actions);
+        }
+      };
+
     return Hooks.runBefore(suite)
       .then(() => runTests(suite.tests, state, actions, reporter))
       .then(() => runSuites(suite.suites, state, actions, reporter))
+      .then(checkSiblings)
       // Ensure we run the after hooks no matter if the tests/suites fail
       .then(runAfterHooks(false), runAfterHooks(true));
   }
 };
 
 export const runSuites = (suites: Suite[], state: RunState, actions: RunActions, reporter: RunReporter): Promise<void> => {
-  return loop(suites, (suite) => runSuite(suite, state, actions, reporter));
-};
\ No newline at end of file
+  return loop(suites, (suite) => {
+
+    const checkSiblings = !suite.parent?.root ? undefined :
```

Comment:
This looks very strange and is hard to read use a if statement then branch with and without the then(checkSiblings). What does then(undefined) do I guess it's a noop.